snapcraft/commands/daemon.start

```diff
@@ -652,8 +652,8 @@ if [ -n "${PID}" ] && [ "$(readlink "/proc/self/ns/mnt")" != "$(readlink "/proc/
 fi
 
 # LXD
-## Check for existing LXDs
-for pid in $(pgrep --euid 0 --full "lxd(\.debug)? --logfile"); do
+## Check for existing LXDs (check for bare lxd and lxd-daemon processes not running via /bin/sh, as well as lxd.debug processes).
+for pid in $(pgrep --euid 0 --full "(^lxd-daemon|^lxd|lxd.debug) --logfile"); do
```

Comment:
Optional change:

```suggestion
for pid in $(pgrep --euid 0 --full "^lxd(|-daemon|\.debug) --logfile"); do
```
Which reads more naturally to me? Starts with `lxd` followed by nothing or `-daemon` or `.debug` suffix followed by `--logfile`.

Reply:
the `^` bit must only be for lxd-daemon and lxd but not lxd.debug because of the way its invoked it will be prefixed with the path.

Reply:
Very true, thanks! Then the only missing bit is escaping the `.` of `.debug` (`\.debug`).